### PR TITLE
time: Typedef time_t to uint64_t if CONFIG_SYSTEM_TIME64 is defined

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
to handle 2038 problem correctly

## Impact
The type of time_t
 
## Testing
Pass CI

### CODE REVIEW

**Comment by pkarashchenko** [comment]
Seems not all parts of the system are ready for this.

**Comment by xiaoxiang781216** [comment]
Yes, let's fix the problem phase by phase.

**Comment by xiaoxiang781216** [comment]
Wait for https://github.com/apache/incubator-nuttx/pull/6783, mark as draft.

**Comment by xiaoxiang781216** [comment]
> Wait for #6783, mark as draft.

@pkarashchenko this patch is ready now.

File: include/time.h [hunk 1]
```diff
@@ -98,7 +98,11 @@
 
 /* Scalar types */
 
+#ifdef CONFIG_SYSTEM_TIME64
+typedef int64_t   time_t;         /* Holds time in seconds */
```
Comment (pkarashchenko) [inline include/time.h:102]: @xiaoxiang781216 is this issue https://github.com/apache/incubator-nuttx/issues/3390 related to this PR?
I mean that for 64 bit we are going with signed while with 32 bit with unsigned

Comment (xiaoxiang781216) [reply include/time.h:102]: most system typedef time_t to signed type(32bit or 64bit):
https://stackoverflow.com/questions/471248/what-is-time-t-ultimately-a-typedef-to
NuttX is a special one which define time_t to unsigned 32bit. I guess the reason is that the unsigned type could double the the range of representation.
But 64bit has enough bit to represent the very very long range human being can expect, so I follow other OS convention to typedef time_t to signed 64bit.

**Review by pkarashchenko — APPROVED** [review]

**Review by masayuki2009 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

File: include/time.h (hunk 1 above)
Comment (yamt) [reply include/time.h:102]: as discussed in https://github.com/apache/nuttx/pull/4693
consistency within nuttx outweighs consistency with other systems.
can you consider to make it unsigned?

**Comment by yamt** [comment]
this change still have the problems pointed out in the older PR.
https://github.com/apache/nuttx/pull/4693#discussion_r731862198
https://github.com/apache/nuttx/pull/4693#issuecomment-946687511

File: include/time.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/time.h:102]: It's fine to use unsigned, could you provide a patch?

**Pull request merged**